src/backends/gpu/kernels/nernst.cu

```diff
@@ -0,0 +1,36 @@
+#include <cstdint>
+
+#include "../nernst.hpp"
+#include "detail.hpp"
+
+namespace arb {
+namespace gpu {
+
+namespace kernels {
+    template <typename T>
+    __global__ void nernst(std::size_t n, int valency, T temperature, const T* Xo, const T* Xi, T* eX) {
+        auto i = threadIdx.x+blockIdx.x*blockDim.x;
+
+        constexpr T RF = 1e3*8.3144598/96485.33289; // factor 1e3 to scale from V -> mV
+        T factor = RF*temperature/valency;
+        if (i<n) {
+            eX[i] = factor*std::log(Xo[i]/Xi[i]);
+        }
+    }
+} // namespace kernels
+
+void nernst(std::size_t n,
+            int valency,
+            fvm_value_type temperature,
+            const fvm_value_type* Xo,
+            const fvm_value_type* Xi,
+            fvm_value_type* eX)
+{
+    constexpr int block_dim = 128;
+    const int grid_dim = impl::block_count(n, block_dim);
+    kernels::nernst<<<grid_dim, block_dim>>>
+        (n, valency, temperature, Xo, Xi, eX);
+}
```

Comment:
As a micro-optimization, I suspect it would be better to have the CPU compute the `RF*temperature/valency` factor and pass it in as a parameter to the kernel when it is enqueued. Thoughts?

Reply:
It might work as a micro-optimization, but I doubt that this will be a bottleneck. If it is a problem, then we could make the optimization.

Otherwise I find it clearer to have the Nernst equation spelt out directly inside the kernel. Less room for mistakes that way.